consumers/campaigner/main.go

```diff
@@ -243,12 +246,44 @@ func processSubscribers(
 	return nil
 }
 
-func setStatusFailed(ctx context.Context, store storage.Storage, campaign *entities.Campaign) error {
+// overwriting the callback func for max attempts reached to insert into campaign failed logs.
+func (h *MessageHandler) LogFailedMessage(m *nsq.Message) {
+	if m == nil {
+		return
+	}
+	msg := new(entities.CampaignerTopicParams)
+	err := json.Unmarshal(m.Body, &msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return
+	}
+	logrus.WithFields(logrus.Fields{
+		"user_id":     msg.UserID,
+		"campaign_id": msg.CampaignID,
+	}).Error("exceeded max attempts for sending the campaign")
+
+	campaign, err := h.s.GetCampaign(msg.CampaignID, msg.UserID)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{"campaign_id": msg.CampaignID, "user_id": msg.UserID}).
+			WithError(err).Error("Failed to get campaign.")
+		return
+	}
+	err = h.s.LogFailedCampaign(campaign, "Exceeded max attempts for preparing the campaign")
+	if err != nil {
+		logrus.WithField("campaign_id", campaign.ID).
+			WithError(err).Error("Failed to store campaign failed log.")
+		return
+	}
+}
+
+// logFailedCampaign updates campaign status to failed & inserts campaign  failed log.
+func logFailedCampaign(ctx context.Context, store storage.Storage, campaign *entities.Campaign, description string) error {
```

Comment:
Maybe we should rename this to some name that is saying that its updating the campaign status

Reply:
Hmm i have comment what's doing i don't know any better name? @gudgl

Reply:
I just think that the main goal here is to update the campaign status to failed